.teamcity/src/subprojects/build/core/ProjectCore.kt

```diff
@@ -6,25 +6,32 @@ import subprojects.build.defaultTimeouts
 
 data class JDKEntry(val name: String, val env: String)
 data class OSEntry(val name: String, val agentString: String, val taskName: String)
-data class JavaScriptEngine(val name: String, val dockerContainer: String)
+data class JSEntry(val name: String, val dockerContainer: String)
 data class OSJDKEntry(val osEntry: OSEntry, val jdkEntry: JDKEntry)
 
 const val junitReportArtifact =  "+:**/build/reports/** => junitReports.tgz"
 const val memoryReportArtifact = "+:**/hs_err* => outOfMemoryDumps.tgz"
 
-val operatingSystems = listOf(
-    OSEntry("macOS", "Mac OS X", "linkDebugTestMacosX64"),
-    OSEntry("Li" +
-        "nux", "Linux", "linkDebugTestLinuxX64"),
-    OSEntry("Windows", "Windows", "linkDebugTestMingwX64"))
-val jdkVersions = listOf(
-    JDKEntry("Java 8", "JDK_18"),
-    JDKEntry("Java 11", "JDK_11"))
-val javaScriptEngines = listOf(
-    JavaScriptEngine("Chrome/Node.js", "stl5/ktor-test-image:latest"))
+val macOS = OSEntry("macOS", "Mac OS X", "linkDebugTestMacosX64")
+val linux = OSEntry("Linux", "Linux", "linkDebugTestLinuxX64")
+val windows = OSEntry("Windows", "Windows", "linkDebugTestMingwX64")
+
+val operatingSystems = listOf(macOS, linux, windows)
+
+val java8 = JDKEntry("Java 8", "JDK_18")
+val java11 = JDKEntry("Java 11", "JDK_11")
+
+val jdkVersions = listOf(java8, java11)
+
+val js = JSEntry("Chrome/Node.js", "stl5/ktor-test-image:latest")
+
+val javaScriptEngines = listOf(js)
+
 val stressTests = listOf(
-    OSJDKEntry(OSEntry("Linux", "Linux", "linkDebugTestLinuxX64"), JDKEntry("Java 8", "JDK_18")),
-    OSJDKEntry(OSEntry("Windows", "Windows", "linkDebugTestMingwX64"), JDKEntry("Java 8", "JDK_18")))
+    OSJDKEntry(linux, java8),
+    OSJDKEntry(windows, java8))
+
+val generatedBuilds = hashMapOf<String, BuildType>()
```

Comment:
You put values into this map in different places. Can you fill it just after it's initialization to have related code in one place?

Reply:
But the code that fills in the publishing is relevant to release project, and not to core.

Reply:
Then can you just define `generatedBuilds` it there?